apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,195 @@
 
 type Props = {|
   +data: ResultsListItemType,
 |};
 
-const ResultListItem = (props: Props) => {
-  if (props.data == null) {
+export type Route = $PropertyType<ResultsListItemType, 'route'>;
+export type RouteItem = $ElementType<$NonMaybeType<Route>, number>; // number because arrays are number-indexed
+
+// @TODO get from config
+const dateFormat = 'ddd D MMM';
+
+class ResultListItem extends React.Component<Props> {
+  parseDate = (date: ?Date) => {
+    if (date) return DateFNS.parse(date);
     return null;
-  }
-  const { flyFrom, flyTo, localDeparture, localArrival, price } = props.data;
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem) => {
+    const timeSimpleFormat = 'H:mm';
+
+    const localArrival = this.parseDate(route?.arrival?.localTime);
+    const localDeparture = this.parseDate(route?.departure?.localTime);
+    const utcArrival = this.parseDate(route?.arrival?.utcTime);
+    const utcDeparture = this.parseDate(route?.departure?.utcTime);
+    const duration =
+      route &&
+      `${DateFNS.differenceInHours(
+        utcArrival,
+        utcDeparture,
+      )}h ${DateFNS.differenceInMinutes(utcArrival, utcDeparture) % 60}m`;
+
+    return {
+      arrival: (route && route.arrival?.city) ?? '',
+      arrivalTime:
+        (localArrival && DateFNS.format(localArrival, timeSimpleFormat)) ?? '',
+      carrier: { code: (route && route.airline) ?? '', name: '' }, // @TODO: get name of the airline
+      tripDate:
+        (localDeparture && DateFNS.format(localDeparture, dateFormat)) ?? '',
+      departure: (route && route.departure?.city) ?? '',
+      departureTime:
+        (localDeparture && DateFNS.format(localDeparture, timeSimpleFormat)) ??
+        '',
+      duration: duration ?? '',
+      id: (route && route.id) ?? '',
+    };
+  };
+
+  getSectorBorderIndex = () => {
+    const { route, routes } = this.props.data;
```

Comment:
I'm confused about this naming too: is `routes` an array of `route`?

Reply:
It is not. The naming is from REST API and it's not good. We should change it in the future according to @FilipMessa 's conventions. 

`routes` is array of sectors (like `[[OSL, PRG], [PRG, OSL]]`) and `route` is array with info of individual flights (like on `[0]` could be info about OSL -> BCN, on `[1]` could be info about BCN -> PRG, etc.).